pkg/sources/adapter/googlecloudpubsubsource/cloudevents.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2021 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package googlecloudpubsubsource
+
+import (
+	"strings"
+
+	"cloud.google.com/go/pubsub"
+)
+
+const ceExtensionPubSubMessagePrefix = "pubsubmsg"
+
+// ceExtensionAttrsForMessage returns a collection of CloudEvents extension
+// attributes translated from the message attributes of the given Pub/Sub message.
+//
+// The resulting extension attribute name is composed of the 'pubsubmsg' prefix,
+// followed by the lowercase name of the Pub/Sub message attribute, from which all
+// non-alphanumeric characters have been removed (e.g. "pubsubmsgmyattribute").
+//
+// https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#extension-context-attributes
+func ceExtensionAttrsForMessage(msg *pubsub.Message) map[string]interface{} {
+	if len(msg.Attributes) == 0 {
+		return nil
+	}
```

Comment:
The code in this file isn't new either. It was moved from `process.go` and slightly refactored so that the function names are aligned with the ones used in the SQS source (both adapters have a lot in common).